Add AGENTS.md for AI-agent contributors (#1157)
jamulussoftware/jamulussoftware.github.io: contribute/en/Style-and-Tone.md

While contributing to Jamulus or the website, you should also keep style and tone in mind. Have a look at the following guidelines.

> These guidelines apply to every contribution, including work drafted with AI assistance. The `AGENTS.md` file in the website repository points automated contributors here rather than repeating them, so this page stays the single source of truth.

Reviewer: ```suggestion
These guidelines apply to every contribution, including work drafted with AI assistance. 

Note: The `AGENTS.md` file in the website repository points automated contributors here, so this page stays the single source of truth.
```

Reviewer: Please apply this suggestion - the two-sentence version with the "Note:" lead-in reads better and keeps the AGENTS.md reference separate from the applicability statement.

> 🤖 Used AI: opencode, deepseek-v4-flash-free

> These guidelines apply to every contribution, including work drafted with AI assistance. The `AGENTS.md` file in the website repository points automated contributors here rather than repeating them, so this page stays the single source of truth.

Reviewer 2: And I think somewhere around here we need to repeat the comment that's made in CONTRIBUTIONS.md (in jamulussoftware/jamulus) regarding "ownership" - i.e. the _person_ submitting the change must understand what the change is doing and support it.

Plus the AI attribution comment.

"Server List" This is the list of Servers maintained by a Directory.  A Server registers with a Directory to be _listed_ in that Directory’s _server list_.

## Punctuation

> Content here is written for people, but parts of it are read by tools as well. Keep the two apart.

Reviewer: Do we really need this destinction?

Reviewer 2: I don't think "Keep the two apart." is necessarily correct.  I think telling people and AI tools here who the audiences are is useful - by why under "Punctuation" - it's a _non sequeteur_.  That appropriate punctuation and layout can _enable and improve_ reading by both audiences is probably also worth mentioning.

So maybe
```suggestion
Use of text layout, for example lists and punctuation, should aid the reader, whether a person or a tool.
```

Author: I'm not sure our audience wants to wrap its head around machines as readers. I think we agree that the gist of this concern is that SOMETIMES we're writing for a machine, and sometimes it's a human. When I write some code that will be pasted, that's for a machine, and it has real concerns about ASCII compliance. When we're writing for an HTML publication, we need to "code switch" and make sure the paste blocks are paste-ready. And the prose around them is for humans, and I will continue advocating for their richness as compared to machine-ready content.

Notice I didn't mention AI. I'm not sure this is about AI, here. AI can handle itself on the humanist of for-humans HTML.

Content here is written for people, and sometimes those people feed parts of it into machines. I think that's the central thrust intended here.  Put that way, we should hit upon the ease of doing that with aides like click-to-copy, rather than select-and-copy, along with code width concerns (which need to work for both audiences).

I also believe this depth and nuance is better said to an agent, so maybe this guidance belongs in the repo AGENTS.md. I think perhaps we can let people submit PRs that are as sloppy as they want, and then just ask an agent to clean it up by these standards. Why educate people in this dead art? (And I have 25 years of professional experience in this particular dead art!)

Reviewer 2: > I think perhaps we can let people submit PRs that are as sloppy as they want, and then just ask an agent to clean it up by these standards.

Because humans will be reviewing, this needs to happen before submission.  I'm happy to bounce non-compliant PRs immediately back to "needs work", though.

Author: > Because humans will be reviewing, this needs to happen before submission. 

Put an LLM in the CI.

Reviewer: > Put an LLM in the CI.

Better use a specialized tool for AI reviews unless we go "local" on the CI.

Reviewer 2: The Github token allowance for open source projects is limited.  It's better to keep responsibility on the submitter directly.  However, that doesn't mean we shouldn't supply a required set of rules that need complying with -- which is what the AGENTS.md is meant to guide towards.

Github seems to have an agent that happily butts in already, anyway - I don't think we should be developing and maintaining anything additional ourselves.  (With any luck, it'll also read the AGENTS.md.)

Author: I think those are authentication tokens. 

The cost of an LLM varies substantially based on the size of the LLM. This kind of task would suit an affordable model, or even an open-weight one. I presume there are limits on what CI is willing to do for free. But CI need not do anything here. If someone submits a PR, I can review it against AGENTS.md and push fixes in a minute or two. Our documentation pipeline has always seemed unusually tricky. AGENTS.md in this repo should change that.

Reviewer 2: > I think those are authentication tokens.

Uh, no.  They're what I pay $10 a month for to get some extra models over and above the free tier, plus extended usage.  Free tier usage runs out _very_ quickly.
